packages/swingset/src/get-nav-tree.ts

```diff
@@ -1,49 +1,86 @@
-import { ComponentEntity, DocsEntity } from './types.js'
+import {
+  ComponentEntity,
+  DocsEntity,
+  ComponentNode,
+  CategoryNode,
+} from './types.js'
 
 export function getNavigationTree(entities: (ComponentEntity | DocsEntity)[]) {
```

Comment:
Some unit tests for this function could be helpful! It'd make it easier for us to iterate on the logic going forward too. We can do this as a follow-up.